### DISCUSSION TITLE
Best practies of ROM

### DISCUSSION THREAD

**Discussion by jacktang** [pymor/pymor#2019] [Q&A]
Hello,

The tutorial of pyMOR is great! and I read it many times and it helps me a lot. As to snapshots based ROM, I have some questions in parametric rom practices.

1. How to determine how many snapshots needed in training?
2. What should I do if the projection error on training data is large?
3. How to make parametric rom more accurate?

Appreciate your time!

**Comment by pmli** [comment]
Hi @jacktang, sorry for the delayed response. These are good questions, but I don't think there are good general answers.

> 1. How to determine how many snapshots needed in training?

If you have an error measure, you could see how it behaves on testing data (or cross-validation) as the amount of training data is increased. There is still the question of what parameter values to use, and, for time-dependent problems, how many time steps to take and how many POD modes to extract.

> 2. What should I do if the projection error on training data is large?

One approach would be to increase the training set and the size of the reduced-order model. The other approach would be to conclude that a linear projection-based method is not appropriate for the problem and look into nonlinear or non-projection-based methods.

> 3. How to make parametric rom more accurate?

The answer is similar to the second question.

**Comment by jacktang** [comment]
@pmli Thanks for your kindly answer! I thought the questions were too general to get response :)

> If you have an error measure, you could see how it behaves on testing data (or cross-validation) as the amount of training data is increased. 

I was thinking is there any increased way to build ROM? which means I only provide some small amount of training dataset at first, and the model will suggest the parameters whose snapshots should be generate  in next steps.

> There is still the question of what parameter values to use, and, for time-dependent problems, how many time steps to take and how many POD modes to extract.

Yes,  I am facing the issue. The global sensitivity analysis will be included in pymor project?

**Reply by pmli** [reply]
There are ways of selecting parameter values to sample, e.g., using a posteriori error estimators that reduced basis methods are known for. Some are already available in pyMOR (@sdrave can comment on this).

**Reply by jacktang** [reply]
I've got `pymor.algorithms.error` module and read it, thanks @pmli !
